### PR TITLE
Dto reflection

### PR DESCRIPTION
**Pull request by man-zhang**
fix dto schema extraction for external services

### CODE REVIEW

File: client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/SutInfoDto.java
Comment (arcuri82) [inline client-java/controller-api/src/main/java/org/evomaster/client/java/controller/api/dto/SutInfoDto.java:61]: wasn't this code done in another branch? am i looking at a branch that pull in from another branch that has not been merged yet? in that case, i should wait until the other branch is merged before looking into this one

**Review by arcuri82 — CHANGES_REQUESTED** [review]

File: client-java/controller/src/test/java/com/thrift/example/artificial/PrivateFieldInRequestDto.java
Comment (man-zhang) [reply client-java/controller/src/test/java/com/thrift/example/artificial/PrivateFieldInRequestDto.java:110]: this is the case that exists in the industrial case study. the setter of the attribute might employ a bit different type. 
In order to support such a case, now we analyze the type of attribute and setter/getter (see `AccessibleSchema`).
If we do not support this, there will be problems in generated tests (eg, compile failed due to mismatched type).

Comment (arcuri82) [reply client-java/controller/src/test/java/com/thrift/example/artificial/PrivateFieldInRequestDto.java:110]: @man-zhang ah, sorry, didn't realize it was just an example in a test... yes, we need to support all different types. but i guess most important is what defined in the getters, unless the fields are public

**Review by arcuri82 — APPROVED** [review]

**Pull request merged**
